src/core/Character.cpp

```diff
@@ -20,13 +20,35 @@ void Character::update(sf::Time elapsed) {
     move();
 }
 
-void Character::attack(std::vector<std::unique_ptr<Projectile>> &projectiles, sf::Vector2f mousePos, sf::Time elapsed) {
-    if (attackDelay-elapsed<sf::Time::Zero) {
+void Character::attack(std::vector<std::unique_ptr<Projectile>> &projectiles, 
+                        sf::Vector2f mousePos, 
+                        sf::Time elapsed) 
+{
+    // Change direction character is looking in x direction
+    float xDir = getCenter().x - mousePos.x;
+    if ((xDir > 0 && !left) || (xDir < 0 && left)) {
+        look();
+    } 
+
+    if (attackDelay - elapsed >= sf::Time::Zero) {
+        return;
+    }
+
+    // Determine attack based on class
```

Comment:
Maybe later this could be replaced by the prototype design pattern.
Each character could have a Projectile member and then it would be just cloned to produce more projectiles.
I think it would look more elegant than a bunch of ifs, but for now it works fine.